pkg/engine/events.go

```diff
@@ -225,31 +226,37 @@ func makeEventEmitter(events chan<- Event, update UpdateInfo) (eventEmitter, err
 	logging.AddGlobalFilter(logging.CreateFilter(secrets, "[secret]"))
 
 	buffer, done := make(chan Event), make(chan bool)
-	go queueEvents(events, buffer, done)
-
-	return eventEmitter{
-		done: done,
-		ch:   buffer,
-	}, nil
+	closed := new(int32)
+	*closed = 0
+	emitter := eventEmitter{
+		closed: closed,
+		done:   done,
+		ch:     buffer,
+	}
+	go queueEvents(emitter, events, buffer, done)
+	return emitter, nil
 }
 
 func makeQueryEventEmitter(events chan<- Event) (eventEmitter, error) {
 	buffer, done := make(chan Event), make(chan bool)
-
-	go queueEvents(events, buffer, done)
-
-	return eventEmitter{
-		done: done,
-		ch:   buffer,
-	}, nil
+	closed := new(int32)
+	*closed = 0
+	emitter := eventEmitter{
+		closed: closed,
```

Comment:
Is there a reason that would be preferred over this?

```suggestion
	closed := 0
	emitter := eventEmitter{
		closed: &closed,
```